### ISSUE TITLE
Possible confusion with use of arrays in options and in utilities::translateArrayOptions()

### ISSUE THREAD

**Issue by samseaver** [ModelSEED/ModelSEED#153]
1) utilities::translateArrayOptions(), if an array is passed in, returns a 2D arrayref
2) If the '@' symbol is used in the definition for the opt_spec() function, the string is inserted into an arrayref.

The end result is, if I did something like `--mergeto BioCyc,PlantCyc`, `$opts->{mergeto}` would be an arrayref, and passing it into utilties::translateArrayOptions() would return a 2D arrayref that looks like:

```
[[BioCyc],[PlantCyc]]
```

when all I need is:

```
[BioCyc,PlantCyc]
```

So the question is, which one are we going to stick to. We can either use 1) or 2) but not both in the same command.

I should add that this behavior was recently implemented in App::bio::Command::addcpdtable.pm

**Comment by cshenry** [comment]
I disagree. I want both in the same command, and I don't see why that should be a problem.
As far as I know, my code was working… it works in the "list" command at least.

I think this should be fine:
-w "id" -w "name"

And this should be fine:
-w "id,name"

If the code that makes this work is having a problem, well then… let's fix it.
Are you sure it's even broken? Like I said, it works fine in the list command.

**Comment by samseaver** [comment]
The list command does not use the utilities::translateArrayOptions() function, and instead uses this concise line:

```
push(@$with, split(/,/,join(",",@{$opts->{with} || []})));
```

I repeat what I say, we either use the '@' symbol _or_ the translateArrayOptions() function.

**Comment by devoid** [comment]
I think I see what Sam is talking about; here's the function in question:

``` perl
sub translateArrayOptions {
    my ($args) = @_;
    $args = ModelSEED::utilities::ARGS($args,["option"],{
        delimiter => "|"
    });
    if ($args->{delimiter} eq "|") {
        $args->{delimiter} = "\\|";
    }
    my $output = [];
    if (ref($args->{option}) eq "ARRAY") {
        foreach my $item (@{$args->{option}}) {
            push(@{$output},[split($args->{delimiter},$item)]);
        }
    } else {
        $output = [split($args->{delimiter},$args->{option})];
    }
    return $output;
}
```

What Sam wants is to change this line:

``` perl
push(@{$output},[split($args->{delimiter},$item)]);
push(@{$output}, split($args->{delimiter},$item));
```

This seems reasonable to me. Chris, is there a reason it should return `[ [ 'A' ], [ 'B' ] ]`, in place of `[ 'A', 'B' ]` ?

**Comment by cshenry** [comment]
No, I think that's just a bug. The idea was to return all results in exactly the same format.
Chris
